### PR TITLE
Migrations: Make Table to TableBuilder & Create new Table Obj & implement exists()/delete() into Table Object.

### PR DESCRIPTION
**Pull request by sayore**
We can use the same style as in the QueryBuilder to create a table.

+ Added addField({})
+ Changed Tests(for Table)
+ Changed Adapter to load a TableBuilder via tableBuilder()
+ Made Table to Masterobject to handle tables. (Which in turn will produce a TableBuilder which was previously Table)
+ Added Test to Table, Creating and Deleting a Test Table.
+ Added exists() and delete() to Table Object
+ Added size to ColumnDefinition instances. (I ran into errors when VARCHAR was undefined, could be something different, needs to be checked though)
+ Added Stubs for Info and Modify into Table

### CODE REVIEW

File: src/table_test.ts [hunk 1]
```diff
@@ -1,38 +1,74 @@
...
 });
 
 Deno.test("Table: postgres", () => {
-  const table = new Table("users", [
-    { type: "increments", name: "id", primaryKey: true, autoIncrement: true },
-    { type: "varchar", name: "name" },
-  ], { dialect: "postgres" });
+  const table = new TableBuilder("users", undefined, { dialect: "postgres" })
+    .addField(
+      { type: "increments", name: "id", primaryKey: true, autoIncrement: true },
```
Comment (sayore) [inline src/table_test.ts:34]: I don't exactly know why this line get's a comma at the end, deno fmt seems to think this is an array?
Runs well however in any case..

File: src/table.ts [hunk 2]
```diff
@@ -81,4 +148,79 @@ export class Table {
...
+  }
+
+  /** Checks if this Table exists. */
+  async exists(): Promise<boolean> {
+    if (!this.adapter) {
+      throw new Error("Cannot run query, adapter is not provided!");
+    }
+    // SELECT name FROM sqlite_master WHERE type='table' AND name='${tablename}';
+
+    switch (this.adapter.type) {
```
Comment (rahmanfadhil) [inline src/table.ts:197]: Even though this code works, it's pretty hard to read. You need to refactor it into something like this:

```ts
switch (this.adapter.type) {
  case "sqlite":
    const sqliteQuery = await this.adapter.query(
      `SELECT name FROM sqlite_master WHERE type='table' AND name='${this.tableName}'`,
    );
    return sqliteQuery.records.length === 1;
  case "postgres":
    const postgresQuery = await this.adapter.query(
      `SELECT EXISTS (SELECT FROM pg_tables WHERE tablename = '${this.tableName}')`,
    );
    if (postgresQuery.records[0]) {
      let result = postgresQuery.records[0] as any;
      return result.exists;
    } else {
      return false;
    }
  case "mysql":
    const mysqlQuery = await this.adapter.query(
      `SELECT * FROM information_schema.tables WHERE table_name = '${this.tableName}' LIMIT 1`,
    );
    return mysqlQuery.records.length === 1;
  default:
    return false;
}
```

This code works exactly the same as yours.

**Review by rahmanfadhil — CHANGES_REQUESTED** [review]
Thanks for the PR, that's a lot of work you've done!

The code looks good. However, it needs a little bit of refactoring.

Sorry for the late response by the way, I'm pretty busy lately.

File: src/table.ts (hunk 2 above)
Comment (sayore) [reply src/table.ts:197]: Done!

**Comment by sayore** [comment]
Should be good then, feel free to refactor the signatures how you imagine, hope i got some heavy lift done for you 👍
If you want i can add some more functionality too in the coming days(column info, row count info, a check for valid id function, ...), but you can first refactor how you imagine the libary to work! 

Tell me what you need i gotchu fam!

**Review by rahmanfadhil — APPROVED** [review]
Looks great overall!

**Comment by rahmanfadhil** [comment]
> Should be good then, feel free to refactor the signatures how you imagine, hope i got some heavy lift done for you 👍
> If you want i can add some more functionality too in the coming days(column info, row count info, a check for valid id function, ...), but you can first refactor how you imagine the libary to work!
> 
> Tell me what you need i gotchu fam!

Yeah, I think I need to design the migrations API first. Probably making a short markdown doc about how this feature should look like from the outside. I'll let you know immediately when it's ready.

Thanks a lot!

**Pull request merged**